src/types/input.ts

```diff
@@ -0,0 +1,36 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import type { InputHTMLAttributes, ReactElement } from 'react';
+
+export type InputBaseProps = {
+    decorator?: ReactElement;
+    error?: boolean;
+    errorText?: string;
+    helperText?: string;
+    onChange?: (value: string) => void;
```

Comment:
```suggestion
    onChange?: (value: number) => void;
```
wdyt?

Reply:
My opinion is to always return the same type as the input.  But happy to make this change if you feel it is the best one.